### PR TITLE
Issue #69: resolved removeViolationComments Bug

### PR DESCRIPTION
**Pull request by Anmol202005**
Fixes: #69

### CODE REVIEW

**Review by rdiachenko — CHANGES_REQUESTED** [review]
ok, this is the third time you fix it, I'm now asking to cover it by tests, all possible edge cases please

File: src/test/java/org/checkstyle/autofix/RemoveViolationComments.java [hunk 1]
```diff
@@ -49,21 +48,31 @@ public TreeVisitor<?, ExecutionContext> getVisitor() {
...
-                            if (textComment.getText().startsWith("violation")) {
-                                suffixAccumulator.append(textComment.getSuffix());
-                                result = null;
-                            }
+            for (Comment comment : space.getComments()) {
+                Comment result = comment;
+                if (!comment.isMultiline() && comment instanceof TextComment) {
+                    final TextComment textComment = (TextComment) comment;
+                    if (textComment.getText().matches("\\s*(\\d+\\s*)?violation.*")) {
+                        if (!filteredComments.isEmpty()) {
```
Comment (rdiachenko) [inline src/test/java/org/checkstyle/autofix/RemoveViolationComments.java:59]: swap condition, follow this pattern for better code readability: `if (positive condition) {...} else {...}`
This is not the first time I see this.
What's easier to read? `if (filteredComments.isEmpty())` or `if (!filteredComments.isEmpty())`

File: src/test/java/org/checkstyle/autofix/RemoveViolationComments.java [hunk 2]
```diff
@@ -49,21 +48,31 @@ public TreeVisitor<?, ExecutionContext> getVisitor() {
...
+                            filteredComments.set(lastIndex,
+                                    previousComment.withSuffix(textComment.getSuffix()));
                         }
-                        return result; })
-                    .filter(Objects::nonNull)
-                    .collect(Collectors.toList());
+                        else {
+                            suffixAccumulator = textComment.getSuffix();
+                        }
+                        result = null;
```
Comment (rdiachenko) [inline src/test/java/org/checkstyle/autofix/RemoveViolationComments.java:68]: instead of setting it null and later checking for null you may extend the outer if condition with `else { filteredComments.add(result); }`

File: src/test/java/org/checkstyle/autofix/RemoveViolationComments.java (hunk 1 above)
Comment (Anmol202005) [reply src/test/java/org/checkstyle/autofix/RemoveViolationComments.java]: yep, done
will be careful with this.

**Review by rdiachenko — APPROVED** [review]
not ideal, logic is too complex and not modular, but this class is used for tests purposes only and we already spent too much time on it, ok to merge
